pkg/tasks/tasks.go

```
@@ -269,9 +277,13 @@ func (self *ViewBufferManager) NewCmdTask(start func() (*exec.Cmd, io.Reader), p
 			refreshViewIfStale()
 
 			if err := cmd.Wait(); err != nil {
-				// it's fine if we've killed this program ourselves
-				if !strings.Contains(err.Error(), "signal: killed") {
```

When navigating through branches with a lot of commits in the Branches panel, we always enter this error branch. The process will still be running indefinitely due to the buffered git log output so when we run the next git log, the previous process will be killed.

There is a string check here to suppress the failure logs due to this reason but on Windows, the string is different ("exit status 1").

Fix: Check whether we sent a kill signal and if so, suppress the error logs.

---

Now that we've made this change, do we still need to check on `signal: killed`?

---

I saw another usage of oscommands.Kill in RunAndProcessLines when the command decides it's time to stop reading. I think that could be another source of the kill signal here and should still be suppressed in logs on non-Windows where the string match is possible.

---

It took me a while to understand this. There's no further change we should make, neither here nor in RunAndProcessLines, is that right? Ready to merge? This is the last PR that blocks the release, so it would be nice to finish it.

---

Yeah I'm wondering if we should be doing something like this in `RunAndProcessLines`:
```diff
diff --git a/pkg/commands/oscommands/cmd_obj_runner.go b/pkg/commands/oscommands/cmd_obj_runner.go
index 41fedcfbc..7f209cbc4 100644
--- a/pkg/commands/oscommands/cmd_obj_runner.go
+++ b/pkg/commands/oscommands/cmd_obj_runner.go
@@ -166,6 +166,12 @@ func (self *cmdObjRunner) RunAndProcessLines(cmdObj ICmdObj, onLine func(line st
 		return err
 	}
 
+	done := make(chan struct{})
+	go func() {
+		_ = cmd.Wait()
+		close(done)
+	}()
+
 	for scanner.Scan() {
 		line := scanner.Text()
 		stop, err := onLine(line)
@@ -173,17 +179,25 @@ func (self *cmdObjRunner) RunAndProcessLines(cmdObj ICmdObj, onLine func(line st
 			return err
 		}
 		if stop {
-			_ = Kill(cmd)
+			select {
+			case <-done:
+			default:
+				_ = Kill(cmd)
+			}
 			break
 		}
 	}
 
 	if scanner.Err() != nil {
-		_ = Kill(cmd)
+		select {
+		case <-done:
+		default:
+			_ = Kill(cmd)
+		}
 		return scanner.Err()
 	}
 
-	_ = cmd.Wait()
+	<-done
 
 	self.log.Infof("%s (%s)", cmdObj.ToString(), time.Since(t))
 

```

---

Just to clarify my earlier comment, I was not suggesting a change to RunAndProcessLines.

1. In tasks.go, I added a `case <-opts.Stop:` branch so Windows won't produce an error log on every arrow key press in the Branches panel.
2. A question was asked about whether we still need to the string check now that the `case <-opts.Stop:` branch is added.
3. I thought the answer was possibly yes because this stopping logic in tasks.go when you have another queued task is not the only place which can cause commands to be killed. I also saw another usage in RunAndProcessLines. So the string check could be kept so the log suppressing still works for non-Windows in these specific scenarios. On Windows, you're out of luck.
4. Looking at the code again, I think I was just completely mistaken earlier. There should be no cases where an exec.Cmd simultaneously goes through RunAndProcessLines and the function being updated in this PR, it doesn't even make sense as a concept.

I updated the PR to remove the string check. Sorry for the confusion.

---

Makes sense. Nonetheless, do you think my above snippet makes sense (i.e. is it worth me raising a separate PR for that?)

---

Makes sense to me. The command may have finished but the scanner is still reading from its internal buffer when deciding to stop or getting an error.

---

Cool. For the record I tested the above snippet out and had issues with lazygit crashing. Nevertheless, this PR is good to merge.